File: app/src/main/java/com/worldventures/dreamtrips/social/ui/profile/service/analytics/FriendRelationshipAnalyticAction.java
package com.worldventures.dreamtrips.social.ui.profile.service.analytics;

import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.janet.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.BaseAnalyticsAction;

@AnalyticsEvent(action = "friends_activity", trackers = AdobeTracker.TRACKER_KEY)
public class FriendRelationshipAnalyticAction extends BaseAnalyticsAction {

   @Attribute("unfriend") String unfriendAction;
   @Attribute("reject_friend_request") String rejectAction;
   @Attribute("cancel_friend_request") String cancelAction;

   private FriendRelationshipAnalyticAction() {
   }

   public static FriendRelationshipAnalyticAction cancelRequest() {
      FriendRelationshipAnalyticAction action = new FriendRelationshipAnalyticAction();
      action.cancelAction = "1";
      return action;
   }

   public static FriendRelationshipAnalyticAction rejectRequest() {
      FriendRelationshipAnalyticAction action = new FriendRelationshipAnalyticAction();
      action.rejectAction = "1";
      return action;
   }

   public static FriendRelationshipAnalyticAction unfriend() {
      FriendRelationshipAnalyticAction action = new FriendRelationshipAnalyticAction();
      action.unfriendAction = "1";
      return action;
   }

}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/profile/analytics/FriendRelationshipAnalyticAction.kt
package com.worldventures.dreamtrips.social.service.profile.analytics

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.core.service.analytics.BaseAnalyticsAction
import com.worldventures.janet.analytics.AnalyticsEvent

@AnalyticsEvent(action = "friends_activity", trackers = [(AdobeTracker.TRACKER_KEY)])
class FriendRelationshipAnalyticAction private constructor(
      @Attribute("unfriend") var unfriendAction: String? = null,
      @Attribute("reject_friend_request") var rejectAction: String? = null,
      @Attribute("cancel_friend_request") var cancelAction: String? = null
) : BaseAnalyticsAction() {
   companion object {

      @JvmStatic
      fun cancelRequest() = FriendRelationshipAnalyticAction(cancelAction = "1")

      @JvmStatic
      fun rejectRequest() = FriendRelationshipAnalyticAction(rejectAction = "1")

      @JvmStatic
      fun unfriend() = FriendRelationshipAnalyticAction(unfriendAction = "1")
   }
}
